src/components/web-form-renderer.vue

```diff
@@ -170,14 +243,14 @@ const postPrimaryInstance = (file) => {
         submissionModal.show({ type: 'errorModal', errorMessage: data.message });
         return false;
       }
-      instanceId.value = data.instanceId;
+      newInstanceId.value = data.instanceId;
       return true;
     })
     .catch(noop);
 };
 
 const uploadAttachment = async (attachment) => {
-  const url = withToken(apiPaths.submissionAttachment(form.projectId, form.xmlFormId, !form.publishedAt, instanceId.value, attachment.file.name));
+  const url = withToken(apiPaths.submissionAttachment(form.projectId, form.xmlFormId, !form.publishedAt, props.instanceId, attachment.file.name));
```

Comment:
TODO: `props.instanceId` will be null in case of new submission, get it as function argument